target/linux/ramips/mt7620/base-files/etc/board.d/02_network

```diff
@@ -419,6 +423,10 @@ ramips_setup_macs()
 	tplink,archer-mr200)
 		wan_mac=$(macaddr_add "$(mtd_get_mac_binary rom 0xf100)" 1)
 		;;
+	wavlink,wl-wn531g3)
+		wan_mac=$(mtd_get_mac_binary factory 0x2e)
```

Comment:
I'm wondering if we can set the wan MAC via device tree? Please try adding `"mac-address"` nvmem cell to port 4 in dts.

Reply:
Im not sure how to do that. Could you explain it?

Reply:
Please try adding these codes to nvmem-layout node
```
					macaddr_factory_2e: macaddr@2e {
						reg = <0x2e 0x6>;
					};
```
and adding
```
nvmem-cells = <&macaddr_factory_2e>;
nvmem-cell-names = "mac-address";
```
to the port@4 node.

Reply:
It works